specification/app/resource-manager/Microsoft.App/preview/2022-01-01-preview/EasyAuthConfigs.json

```diff
@@ -0,0 +1,706 @@
+        },
+        "registration": {
+          "$ref": "#/definitions/TwitterRegistration",
+          "description": "The configuration settings of the app registration for the Twitter provider."
+        }
+      }
+    },
+    "TwitterRegistration": {
+      "description": "The configuration settings of the app registration for the Twitter provider.",
+      "type": "object",
+      "properties": {
+        "consumerKey": {
+          "description": "The OAuth 1.0a consumer key of the Twitter application used for sign-in.\nThis setting is required for enabling Twitter Sign-In.\nTwitter Sign-In documentation: https://dev.twitter.com/web/sign-in",
+          "type": "string"
+        },
+        "consumerSecretRefName": {
+          "description": "The app secret ref name that contains the OAuth 1.0a consumer secret of the Twitter\napplication used for sign-in.",
+          "type": "string"
+        }
+      }
+    },
+    "LegacyMicrosoftAccount": {
+      "description": "The configuration settings of the legacy Microsoft Account provider.",
+      "type": "object",
+      "properties": {
+        "enabled": {
+          "description": "<code>false</code> if the legacy Microsoft Account provider should not be enabled despite the set registration; otherwise, <code>true</code>.",
+          "type": "boolean"
+        },
+        "registration": {
+          "$ref": "#/definitions/ClientRegistration",
+          "description": "The configuration settings of the app registration for the legacy Microsoft Account provider."
+        },
+        "login": {
+          "$ref": "#/definitions/LoginScopes",
+          "description": "The configuration settings of the login flow."
+        },
+        "validation": {
+          "$ref": "#/definitions/AllowedAudiencesValidation",
+          "description": "The configuration settings of the legacy Microsoft Account provider token validation flow."
+        }
+      }
+    },
+    "Apple": {
+      "description": "The configuration settings of the Apple provider.",
+      "type": "object",
+      "properties": {
+        "enabled": {
+          "description": "<code>false</code> if the Apple provider should not be enabled despite the set registration; otherwise, <code>true</code>.",
+          "type": "boolean"
+        },
+        "registration": {
+          "$ref": "#/definitions/AppleRegistration",
+          "description": "The configuration settings of the Apple registration."
+        },
+        "login": {
+          "$ref": "#/definitions/LoginScopes",
+          "description": "The configuration settings of the login flow."
+        }
+      }
+    },
+    "AppleRegistration": {
+      "description": "The configuration settings of the registration for the Apple provider",
+      "type": "object",
+      "properties": {
+        "clientId": {
+          "description": "The Client ID of the app used for login.",
+          "type": "string"
+        },
+        "clientSecretRefName": {
```

Comment:
>clientSecretRefName [](http://example.com/codeflow?start=9&length=19)

Add x-ms-secret extension if this is a secret.

Reply:
Not secret as above